### PR TITLE
Add read latest 311 45

### LINKED ISSUE #45: Intent: Read latest reports to Boston 311
User should be able to ask what the latest N reports to 311 are.

We can get this info from https://311.boston.gov/. We can also look into using [Open 311](https://mayors24.cityofboston.gov/open311)

### PR DESCRIPTION
**Pull request by jmartini**
Closes #45 

Adds an intent to read the latest 311 reports. Defaults to 3 reports, but user can specify a particular number if they want.

### CODE REVIEW

File: mycity/mycity/intents/latest_311_intent.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+def get_raw_311_reports_json(number_entries):
+    """
+    Returns the JSON object from the 311 API
+
+    :param number_entries: Number of entries to return
+    :return: JSON object containing 311 data
+    """
+    data_url = "https://data.boston.gov/api/3/action/datastore_search"
+    parameters = {
+        "resource_id": "2968e2c0-d479-49ba-a884-4ef523ada3c0",
```
Comment (curranjm) [inline mycity/mycity/intents/latest_311_intent.py]: Can we put this in a constant?

File: mycity/mycity/intents/latest_311_intent.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+from mycity.intents.speech_constants.latest_311_constants import *
+
+DEFAULT_NUMBER_OF_REPORTS = 3
+
+
+def get_311_requests(mycity_request):
+    """
+    Generates response object for a 311 request inquiry.
+
+    :return: MyCityResponseDataModel object
```
Comment (jaumb) [inline mycity/mycity/intents/latest_311_intent.py]: Only one very minor comment: `:param mycity_request: MyCityRequestDataModel object` documentation missing here

**Comment by curranjm** [comment]
Added one comment about putting the 311 api resource id in a constant, but otherwise looks good. Pulled it down and tested successfully. We probably want to pad out the sample utterances for this one as I tried a couple that didn't work before looking at the intent model to see what you had there.

File: mycity/mycity/intents/latest_311_intent.py (hunk 2 above)
Comment (jmartini) [reply mycity/mycity/intents/latest_311_intent.py]: Added!

File: mycity/mycity/intents/latest_311_intent.py (hunk 1 above)
Comment (jmartini) [reply mycity/mycity/intents/latest_311_intent.py]: Good call. Done

**Comment by jmartini** [comment]
> Added one comment about putting the 311 api resource id in a constant, but otherwise looks good. Pulled it down and tested successfully. We probably want to pad out the sample utterances for this one as I tried a couple that didn't work before looking at the intent model to see what you had there.

@alidaniell Can you think of some more sample utterances to add here?

**Comment by alidaniell** [comment]
@jmartini, here are some more sample utterances to consider adding:

What's the three one one
What three one one
What's the latest three one one
What's the latest three one ones
What's the most recent three one one
What are three one one
What are three one ones
What three one ones

Get me the three one one
Get me the latest three one one
Get me the latest three one ones

Get the latest three one one
Get latest three one one
Get latest three one ones
Get recent three one one
Get recent three one ones
Get me the most recent three one one
Get me the most recent three one ones


Give me the latest three one one
Give me latest three one ones
Give me the most recent three one one
 
Give the three one one 
Give the latest three one one
Give latest three one one
Give latest three one ones
Give the most recent three one one
Give most recent three one one

I want the three one one
I want the latest three one one
I want the latest three one ones
I want latest three one ones
I want three one one
I want three one ones
I want the most recent three one one
I want most recent three one ones

I need the three one one
I need the latest three one one
I need the latest three one ones
I need latest three one ones
I need three one one
I need three one ones
I need the most recent three one one
I need most recent three one ones

report the three one one
report latest three one one
report latest three one ones
report three one ones
report most recent three one one
report most recent three one ones

Request the three one one
Request three one one
Request three one ones
Request latest three one one
Request latest three one ones
Request most recent three one one

Requesting the three one one
Requesting three one one
Requesting three one ones
Requesting latest three one one
Requesting latest three one ones
Requesting most recent three one one

Tell me the three one one 
Tell me the latest three one one
Tell me latest three one ones
Tell me latest three one one
Tell me the most recent three one one

Tell the three one one 
Tell the latest three one one
Tell latest three one one
Tell latest three one ones
Tell the most recent three one one
Tell most recent three one one
Tell most recent three one ones

Say the three one one 
Say three one one
Say the latest three one one
Say the latest three one ones
Say latest three one one
Say latest three one ones
Say the most recent three one one
Say most recent three one one
Say most recent three one ones

Ask for the three one one 
Ask for the latest three one one
Ask for latest three one one
Ask for the latest three one ones
Ask for the most recent three one one

Ask the three one one 
Ask three one one
Ask the latest three one one
Ask the latest three one ones
Ask latest three one one
Ask latest three one ones
Ask the most recent three one one

Top three one one
Top three one ones

**Pull request merged**